test/controllers/register/trustees/individual/TrusteeNameControllerSpec.scala

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.register.trustees.individual
+
+import controllers.ControllerSpecBase
+import controllers.actions._
+import forms.register.PersonNameFormProvider
+import identifiers.register.trustees.individual.TrusteeNameId
+import identifiers.register.trustees.{IsTrusteeNewId, TrusteesId}
+import models.person.PersonName
+import models.{Index, NormalMode}
+import org.mockito.Matchers.any
+import org.mockito.Matchers.{eq => eqTo, _}
+import org.mockito.Mockito.{reset, times, verify, when}
+import org.scalatest.mockito.MockitoSugar
+import play.api.data.Form
+import play.api.libs.json.Json
+import play.api.mvc.Call
+import play.api.test.Helpers._
+import services.UserAnswersService
+import utils.{FakeNavigator, SectionComplete, UserAnswers}
+import views.html.register.trustees.individual.trusteeName
+
+import scala.concurrent.Future
+
+class TrusteeNameControllerSpec extends ControllerSpecBase {
+
+  import TrusteeNameControllerSpec._
+
+  def controller(dataRetrievalAction: DataRetrievalAction = getMandatoryEstablisherCompany): TrusteeNameController =
```

Comment:
Following up on the action from the last tech talks, could we use the app injector here and override the bindings for the bare basics like the DataRetrievalAction?

Reply:
Done!